src/components/Header/HeaderControls.tsx

```diff
@@ -0,0 +1,118 @@
+/*
+MIT License
+
+Copyright (c) 2021-present, Elastic NV
+
+Permission is hereby granted, free of charge, to any person obtaining a copy
+of this software and associated documentation files (the "Software"), to deal
+in the Software without restriction, including without limitation the rights
+to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+copies of the Software, and to permit persons to whom the Software is
+furnished to do so, subject to the following conditions:
+
+The above copyright notice and this permission notice shall be included in
+all copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+THE SOFTWARE.
+*/
+
+import {
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiButton,
+  EuiHealth,
+  useEuiTheme,
+} from "@elastic/eui";
+import React, { useContext } from "react";
+import { Setter } from "../../common/types";
+import { RecordingContext } from "../../contexts/RecordingContext";
+import { TestContext } from "../../contexts/TestContext";
+import { ControlButton } from "../ControlButton";
+import { SaveCodeButton } from "../ExportScriptButton";
+import { TestButton } from "../TestButton";
+
+interface IRecordingHealth {
+  isPaused: boolean;
+  isRecording: boolean;
+}
+
+function RecordingHealth({ isPaused, isRecording }: IRecordingHealth) {
+  if (isRecording && !isPaused)
+    return <EuiHealth color="danger">Recording</EuiHealth>;
+  if (isRecording && isPaused)
+    return <EuiHealth color="warning">Recording paused</EuiHealth>;
+  return <EuiHealth color="subdued">Not recording</EuiHealth>;
+}
+
+interface IHeaderControls {
+  hasActions: boolean;
+  setIsCodeFlyoutVisible: Setter<boolean>;
+}
+
+export function HeaderControls({
+  hasActions: disabled,
+  setIsCodeFlyoutVisible,
+}: IHeaderControls) {
+  const { euiTheme } = useEuiTheme();
+  const { isPaused, isRecording, toggleRecording } =
```

Comment:
These types does not exist in recording context. Results in build error. 

```js
Property 'isPaused' does not exist on type 'IRecordingContext'.  TS2339

    61 | }: IHeaderControls) {
    62 |   const { euiTheme } = useEuiTheme();
  > 63 |   const { isPaused, isRecording, toggleRecording } =
       |           ^
    64 |     useContext(RecordingContext);
    65 |   const { onTest } = useContext(TestContext);
    66 |   return (
```

Reply:
Happened because of a recent rebase, [fixed](https://github.com/elastic/synthetics-recorder/pull/107/commits/48708235ca4e00cd81bc620bf459962b8246e34b).